bench-tps/src/bench.rs

```diff
@@ -402,29 +402,33 @@ fn poll_blockhash<T: Client>(
     client: &Arc<T>,
     id: &Pubkey,
 ) {
-    let mut blockhash_time;
+    let mut blockhash_last_updated = Instant::now();
     loop {
-        blockhash_time = Instant::now();
-        loop {
+        let blockhash_updated = {
             let old_blockhash = *blockhash.read().unwrap();
             if let Ok((new_blockhash, _fee)) = client.get_new_blockhash(&old_blockhash) {
                 *blockhash.write().unwrap() = new_blockhash;
-                break;
+                blockhash_last_updated = Instant::now();
+                true
             } else {
-                if blockhash_time.elapsed().as_secs() > 30 {
-                    panic!("Blockhash is not updating");
+                if blockhash_last_updated.elapsed().as_secs() > 30 {
```

Comment:
Can we just make this longer. I don't think it makes sense to wait forever for a new blockhash.

Reply:
The current behaviour is that a panic in this thread (blockhash poller) won't stop the bench-tps client. I removed the panic for that reason.

Sounds like you are proposing that we wait longer and bubble the panic up to the main thread of the client?

Reply:
Here's the new behaviour in action:
```[2019-12-19T17:48:18.601005450Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:24.035548543Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:29.460499842Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:34.889351170Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:40.336911050Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:45.766780165Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:51.189155125Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:48:56.618533149Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:49:37.689016714Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:49:43.113283315Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:49:48.551284205Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:49:53.988554476Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:49:59.416248033Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:50:04.844861721Z ERROR solana_bench_tps::bench] Blockhash is not updating
[2019-12-19T17:50:10.285160984Z ERROR solana_bench_tps::bench] Blockhash is not updating
```

Reply:
Yes. the client should `exit` if it's not getting a blockhash after some time. Maybe 2 minutes. I didn't think about the panic not exiting the client.